blog/2024-11-linked-signal/README.md

```diff
@@ -0,0 +1,244 @@
+---
+title: 'Angular 19: Introducing LinkedSignal for Responsive Local State Management'
+author: Johannes Hoppe
+mail: team@angular.schule
+published: 2024-11-04
+lastModified: 2024-11-04
+keywords:
+  - Angular
+  - JavaScript
+  - Signals
+  - Reactive Programming
+  - LinkedSignal
+  - Angular 19
+  - Computed Signals 
+language: en
+thumbnail: ???.jpg
+sticky: false
+---
+
+
+In Angular 19, there's a [new experimental feature](https://github.com/angular/angular/commit/8311f00faaf282d1a5b1ddca29247a2fba94a692) called a **Linked Signal** that makes it easier to keep track of local state that depends on other signals. 
+
+It lets us create a writable signal that can automatically reset based on changes in other signals. 
+This makes it particularly useful for situations where local state needs to stay in sync with dynamic data. 
+Here’s a look at what the Linked Signal is, how it works, and some common use cases.
+
+
+## Contents
+
+* [What is LinkedSignal?](/blog/2024-11-linked-signal#what-is-linkedSignal)
+* [Basic Usage of LinkedSignal](/blog/2024-11-linked-signal#basic-usage-of-linkedSignal)
+* [Advanced Scenarios for LinkedSignal](/blog/2024-11-linked-signal#advanced-scenarios-for-linkedSignal)
+* [LinkedSignal vs. Other Signals](/blog/2024-11-linked-signal#linkedSignal-vs-other-signals)
+* [Best Practices for Using LinkedSignal](/blog/2024-11-linked-signal#best-practices-for-using-linkedSignal)
+* [Conclusion](/blog/2024-11-linked-signal#conclusion)
+
+
+<!-- > **🇩🇪 This article is available in German here: [Angular 19: Einführung von LinkedSignal für die lokale reaktionsfähige Zustandsverwaltung](https://angular-buch.com/blog/2024-11-linked-signal)** -->
+
+
+## What is LinkedSignal?
+
+A Linked Signal is an experimental feature that Angular 19 introduces to help you manage state that automatically syncs with other signals. 
+In simple terms, we receive a writable signal that resets itself when the value of its source signal changes.
+A Linked Signal can be created by using the [`linkedSignal()` factory function](https://next.angular.dev/api/core/linkedSignal).
+
+
+A Linked Signal has the following characteristics:
+
+- **Writable and Reactive**: Like a [`signal`](https://angular.dev/guide/signals#writable-signals), we can update the value of a Linked Signal manually, but it also responds to changes in its source.
+- **A Combination of Signal and Computed**: It’s like [`computed`](https://angular.dev/guide/signals#computed-signals) because it derives its value from other signals, but it stays writable, allowing us to override it when needed.
+
```

Comment:
I'm missing a dry example (here or at the beginning of the next section) to understand the API.
You never mention that we don't have to pass in a `source`, but we can use it like `computed()`:

```ts
const timestampMs = signal(Date.now());

// with computed(): not writable
const timestampSeconds = computed(() => timestampMs / 1000);
timestampSeconds.set(0); // ❌ ERROR

const timestampSecondsLinked = linkedSignal(() => timestampMs / 1000);
timestampSecondsLinked.set(0); // ✅
```

Reply:
Signals created with `computed()` or `linkedSignal()` as seen in this example, always update their value when any of the bound signals changes. If we want to restrict the computation to a specific trigger signal, we can specify a `source`. The following Linked Signal only recomputes its value when the source changes:

```
const timestampSeconds = computed({
  source: timestampMs,
  computation: ms => ms / 1000
});
```

Reply:
Now why is `source`+`computation` "better" than just a computation function? Are there cases where we need the one or the other?

Reply:
Interesting: If we just specify a function, this will be the source (shorthand mode). If we specify both source and computation, source will be source, and the computation will be an additional thing. At least that's how I interpret the implementation: https://github.com/angular/angular/blob/main/packages/core/src/render3/reactivity/linked_signal.ts#L131-L142

Reply:
Some thoughts on why `computation` could be separate:
- The computation function also receives the previous value. If we need it, `source` is not enough.
- If we don't even need the source value but only use the source as trigger, separate computation is prettier. `() => 1` instead of `() => this.value(); return 1`
- separation looks nicer. We see where the data comes from and we have a separate function to calculate the new data.
- The two signatures of `linkedSignal` call the source function `computation` (shorthand) and `source` (options object). This looks like can be used as the same and make no big technical difference?